eventuals/catch.h

```diff
@@ -25,6 +25,7 @@ struct _Catch final {
 
     template <typename... Args>
     void Fail(Args&&... args) {
+      static_assert(sizeof...(args) > 0, "Calling 'Fail' with no error");
```

Comment:
What if we just did the following everywhere instead?
```suggestion
    template <typename Error>
    void Fail(Error&& error) {
```

Reply:
In the future, we can provide multiple error types for `ErrorFrom`, we can imagine that there could be multiple types for `Fail`. If we do like this `Fail(Error&& error)`, I believe that we need to rewrite other functions like 'Start', 'Ended', etc.

By the way, I thought a bit about the next iteration for `Composable` and I believe that for `ErrorFrom` we need to provide multiple types, but only one type for 'ValueFrom'.

Reply:
It totally makes sense that we'll want multiple error _types_, but not multiple errors. And if someone does want to send multiple errors then they can just send them as a `std::tuple` and then people can `std::apply` to unpack themselves, or use our `Unpack()` from `eventuals/unpack.h`.

We really simplify the code when we know we always have a single error, so let's do it!